bridge/opentracing/bridge_test.go

```diff
@@ -574,3 +574,101 @@ func TestBridgeSpanContextPromotedMethods(t *testing.T) {
 		assert.True(t, spanContext.(spanContextProvider).HasTraceID())
 	})
 }
+
+func TestBridgeCarrierBaggagePropagation(t *testing.T) {
+	carriers := []struct {
+		name    string
+		carrier interface{}
+		format  ot.BuiltinFormat
+	}{
+		{
+			name:    "TextMapCarrier",
+			carrier: ot.TextMapCarrier{},
+			format:  ot.TextMap,
+		},
+		{
+			name:    "HTTPHeadersCarrier",
+			carrier: ot.HTTPHeadersCarrier{},
+			format:  ot.HTTPHeaders,
+		},
+	}
+
+	testCases := []struct {
+		name             string
+		baggageItems     []bipBaggage
+		wantBaggageItems []bipBaggage
+	}{
+		{
+			name: "single baggage item",
+			baggageItems: []bipBaggage{
+				{
+					key:   "foo",
+					value: "bar",
+				},
+			},
+			wantBaggageItems: []bipBaggage{
```

Comment:
nit: seems redundant. The contract of the Baggage API is if you set a key/value it either causes an error because they are malformed, or they are accepted and returned as-is via get. There's no situation where they should be different, so having this presumption in the tests is misleading too.

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/4776/commits/b4c11fea9b9e2e00df2c4e80dfd7c3a3c89021f5